proto/src/constants.rs

```diff
@@ -43,9 +43,33 @@ pub const DEFAULT_LDAP_LOCALHOST: &str = "localhost:636";
 pub const DEFAULT_REPLICATION_ADDRESS: &str = "127.0.0.1:8444";
 pub const DEFAULT_REPLICATION_ORIGIN: &str = "repl://localhost:8444";
 
-/// Default replication poll window in seconds.
+/// Default replication poll interval in seconds.
 pub const DEFAULT_REPL_TASK_POLL_INTERVAL: u64 = 15;
 
+/// Minimum replication poll interval in seconds:
+pub const MIN_REPL_TASK_POLL_INTERVAL: u64 = 3;
+
+// Estimate of max write ops that can be processed in the corresponding replication poll interval
+pub const MAX_WRITE_THROUGHPUT_BY_TASK_POLL_INTERVAL: [(u64, u64); 13] = [
+    (2945, 3),
+    (4081, 4),
+    (5294, 5),
+    (6351, 6),
+    (7408, 7),
+    (8280, 8),
+    (9105, 9),
+    (10476, 10),
+    (11919, 11),
+    (12395, 12),
+    (12770, 13),
+    (14293, 14),
+    (15893, 15),
+];
+
+// This represents the percentage of the current MAX_WRITE_THROUGHPUT (determined by the current replication interval)
+// above which we will not trigger replication as we are too close to the maximum throughput and we risk to lower the server performance
+pub const REPLICATION_TRIGGER_SAFETY_THRESHOLD: f64 = 0.8;
```

Comment:
So we're not going over 0.8% throughput? I'm guessing you mean the *ratio* of `MAX_WRITE_THROUGHPUT` rather than percentage?

Reply:
Yess it's definitely not 0.8% :D. It should say ratio instead.